marlowe-playground-client/src/Marlowe/Linter.purs

```diff
@@ -675,25 +697,28 @@ lintBounds restAreInvalid t@(Term (Bound l h) pos) = do
   when isInvalidBound $ addWarning InvalidBound t pos
   pure (isInvalidBound && restAreInvalid)
 
-lintAction :: LintEnv -> Term Action -> CMS.State State Effect
+lintAction :: LintEnv -> Term Action -> CMS.State State (Effect /\ Boolean) -- Booleans says whether the action is reachable
```

Comment:
I'd be tempted to `newtype Reachable = Reachable Boolean`. You would have to add some `wrap` and `unwrap` in places but probably not many and you would get clarity, the Monoid and some minor extra safety in return.

Reply:
We can wrap both `Effect` and `Boolean` in a record if it gets more complicated than this. But I think I will rather leave it for now, may make things worse